core/shared/src/main/scala/zio/stm/ZSTM.scala

```diff
@@ -379,7 +391,16 @@ final class ZSTM[-R, +E, +A] private[stm] (
 
   /**
    * Repeats this effect forever (until the first error).
-   */
+   * '''WARNING''': `forever` uses a busy loop to repeat the effect and will consume a
+   * thread until it fails. This is because STM describes a single atomic transaction, and so must either complete,
+   * retry or fail a transaction before yielding back to the ZIO Runtime.
+   * - Ensure repeating the STM effect will eventually fail.
+   * - Consider using the Blocking thread pool for execution of the transaction.
+   */
+  @deprecated(
+    "Repeating until failure doesn't make sense in the context of STM because it will always roll back the transaction",
+    "1.0.2"
```

Comment:
This should be the version that the deprecated method will be removed, so `2.0.0`.

Reply:
According to the scaladoc for `@deprecated` it's when it was first deprecated. If zio does things differently that's fine but this should probably be documented somewhere.  I can't find any mention of this in the repo. Do you still want me to change it?

```scala
/**
*...
*@param  since   a string identifying the first version in which the definition was deprecated
*...
*/
class deprecated(message: String = "", since: String = "") extends scala.annotation.StaticAnnotation
```

Reply:
You're right.